sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileStatusCache.scala

```diff
@@ -49,7 +52,9 @@ object FileStatusCache {
           session.sqlContext.conf.metadataCacheTTL
         )
       }
-      sharedCache.createForNewClient()
```

Comment:
reading the code more, I think this `FileStatusCache` is not per session, but per invocation. If we really want to clear all states for a session, we should keep a map of `Map[String, Seq[FileStatusCache]]`